pkg/controller/pod_volume_restore_controller_legacy.go

```diff
@@ -281,15 +282,25 @@ func (c *PodVolumeRestoreReconcilerLegacy) OnDataPathCompleted(ctx context.Conte
 	// Create the .velero directory within the volume dir so we can write a done file
 	// for this restore.
 	if err := os.MkdirAll(filepath.Join(volumePath, ".velero"), 0755); err != nil {
-		_, _ = c.errorOut(ctx, &pvr, err, "error creating .velero directory for done file", log)
+		errMsg := "error creating .velero directory for done file"
+		if strings.Contains(err.Error(), "no space left on device") {
```

Comment:
1. This error message is linux-only, it may not work on Windows.
2. `--write-sparse-files` is only helpful when the PV has sparse files  during backup, so it may not fix the problem, therefore, the message can be misleading, IMO only documenting it is good enough.